esp-hal/src/system.rs

```diff
@@ -108,173 +82,6 @@ pub enum Peripheral {
-    }
-
-    /// Resets this software-interrupt
-    pub fn reset(&self) {
-        #[cfg(not(any(esp32c6, esp32h2)))]
-        let system = unsafe { &*SYSTEM::PTR };
-        #[cfg(any(esp32c6, esp32h2))]
-        let system = unsafe { &*crate::peripherals::INTPRI::PTR };
-
-        match NUM {
-            0 => {
-                system
-                    .cpu_intr_from_cpu_0()
-                    .write(|w| w.cpu_intr_from_cpu_0().clear_bit());
-            }
-            1 => {
-                system
-                    .cpu_intr_from_cpu_1()
-                    .write(|w| w.cpu_intr_from_cpu_1().clear_bit());
-            }
-            2 => {
-                system
-                    .cpu_intr_from_cpu_2()
-                    .write(|w| w.cpu_intr_from_cpu_2().clear_bit());
-            }
-            3 => {
-                system
-                    .cpu_intr_from_cpu_3()
-                    .write(|w| w.cpu_intr_from_cpu_3().clear_bit());
-            }
-            _ => unreachable!(),
-        }
-    }
-
-    /// Unsafely create an instance of this peripheral out of thin air.
-    ///
-    /// # Safety
-    ///
-    /// You must ensure that you're only using one instance of this type at a
-    /// time.
-    #[inline]
-    pub unsafe fn steal() -> Self {
-        Self
-    }
-}
-
-impl<const NUM: u8> crate::peripheral::Peripheral for SoftwareInterrupt<NUM> {
-    type P = SoftwareInterrupt<NUM>;
-
-    #[inline]
-    unsafe fn clone_unchecked(&mut self) -> Self::P {
-        Self::steal()
-    }
-}
-
-impl<const NUM: u8> crate::private::Sealed for SoftwareInterrupt<NUM> {}
-
-impl<const NUM: u8> InterruptConfigurable for SoftwareInterrupt<NUM> {
-    fn set_interrupt_handler(&mut self, handler: crate::interrupt::InterruptHandler) {
-        SoftwareInterrupt::set_interrupt_handler(self, handler);
-    }
-}
-
-/// This gives access to the available software interrupts.
-#[cfg_attr(
-    multi_core,
-    doc = r#"
-Please note: Software interrupt 3 is reserved
-for inter-processor communication when the `embassy`
-feature is enabled."#
-)]
-#[non_exhaustive]
-pub struct SoftwareInterruptControl {
-    pub software_interrupt0: SoftwareInterrupt<0>,
-    pub software_interrupt1: SoftwareInterrupt<1>,
-    pub software_interrupt2: SoftwareInterrupt<2>,
-    #[cfg(not(all(feature = "embassy", multi_core)))]
-    pub software_interrupt3: SoftwareInterrupt<3>,
-}
-
-impl SoftwareInterruptControl {
-    fn new() -> Self {
-        SoftwareInterruptControl {
-            software_interrupt0: SoftwareInterrupt {},
-            software_interrupt1: SoftwareInterrupt {},
-            software_interrupt2: SoftwareInterrupt {},
-            // the thread-executor uses SW-INT3 when used on a multi-core system
-            // we cannot easily require `software_interrupt3` there since it's created
-            // before `main` via proc-macro so we  cfg it away from users
-            #[cfg(not(all(feature = "embassy", multi_core)))]
```

Comment:
Hmm, I guess this hasn't been working correctly as the embassy feature on esp-hal no longer exists... :sweat_smile:

Reply:
Not sure what to do about this, and probably should be solved separately anyway.

Reply:
Opened https://github.com/esp-rs/esp-hal/issues/1987